packages/xo-server-sdn-controller/src/index.js

```diff
@@ -701,6 +992,133 @@ class SDNController extends EventEmitter {
+      return
+    }
+
+    const client = find(
+      this._ovsdbClients,
+      client => client.host.$ref === poolNetwork.starCenter
+    )
+    if (client === undefined) {
+      return
+    }
+
+    const centerNetwork = centerClient.host.$xapi.getObjectByRef(
+      centerPoolNetwork.network
+    )
+    const remoteNetwork = client.host.$xapi.getObjectByRef(poolNetwork.network)
+    try {
+      await centerClient.resetForNetwork(
+        centerNetwork.uuid,
+        centerNetwork.name_label,
+        true,
+        remoteNetwork.uuid
+      )
+    } catch (error) {
+      log.error('Error while clearing pool center from network', {
+        error,
+        network: remoteNetwork.name_label,
+        host: client.host.name_label,
+        remotePool: remoteNetwork.$pool.name_label,
+        uuid: crossPoolNetworkUuid,
+      })
+    }
+  }
+
+  async _resetCrossPoolConnections(poolNetwork, uuid) {
+    const client = find(
+      this._ovsdbClients,
+      client => client.host.$ref === poolNetwork.starCenter
+    )
+    if (client === undefined) {
+      return
+    }
+
+    const network = client.host.$xapi.getObjectByRef(poolNetwork.network)
+    try {
+      await client.resetForNetwork(network.uuid, network.name_label, true)
+    } catch (error) {
+      log.error('Error while resetting cross pool connections', {
+        error,
+        network: network.name_label,
+        host: client.host.name_label,
+        uuid,
+      })
+    }
+  }
+
+  async _connectNetworks(poolNetwork, centerPoolNetwork, uuid) {
+    const client = find(
+      this._ovsdbClients,
+      client => client.host.$ref === poolNetwork.starCenter
+    )
+    if (client === undefined) {
+      return
+    }
+
+    const centerClient = find(
+      this._ovsdbClients,
+      client => client.host.$ref === centerPoolNetwork.starCenter
+    )
+    const centerNetwork = centerClient.host.$xapi.getObjectByRef(
+      centerPoolNetwork.network
+    )
+    const network = client.host.$xapi.getObjectByRef(poolNetwork.network)
+
+    // Use centerNetwork VNI by convention
+    const { encapsulation = 'gre', vni = '0' } = centerNetwork.other_config
+    try {
+      await client.addInterfaceAndPort(
+        network.uuid,
+        network.name_label,
+        centerClient.host.address,
+        encapsulation,
+        vni,
+        centerNetwork.uuid
+      )
+      await centerClient.addInterfaceAndPort(
+        centerNetwork.uuid,
+        centerNetwork.name_label,
+        client.host.address,
+        encapsulation,
+        vni,
+        network.uuid
+      )
+    } catch (error) {
+      log.error('Error while connecting networks', {
+        error,
+        network: network.name_label,
+        host: client.host.name_label,
+        pool: client.host.$pool.name_label,
+        centerHost: centerClient.host.name_label,
+        centerPool: centerClient.host.$pool.name_label,
+        uuid,
+      })
+    }
+    log.debug('Networks connected', {
```

Comment:
I don't get it, so you're logging that the networks are connected even if it failed?

Reply:
Yes, but I shouldn't :)
I'm changing it.

Reply:
Has it been changed?

Reply:
Yes, I added a `return` in the error case.